modules/ank/arrow-ank/src/main/kotlin/arrow/ank/interpreter.kt

```diff
@@ -150,7 +150,7 @@ val interpreter: AnkOps = object : AnkOps {
               } else { // accumulating code inside a fence
                 val modifiedSnippet = state.snippet.copy(
                   fence = state.snippet.fence + "\n" + line,
-                  code = state.snippet.code + "\n" + line
+                  code = state.snippet.code + (if (state.snippet.code.isEmpty()) "" else "\n") + line
```

Comment:
Does someone know why we don't have snippets in arrow with an extra newline at the start? This sure looks like there should be one. Is it because of the old ank gradle version (0.7.1)?

Reply:
On current master the output actually contains this one excessive newline, which is fixed with this change. Is there another processing step removing this somehow, because those don't show up on the docs site?

Reply:
I am not able to generate the site at the moment :disappointed:, so I can't really check what Ank produces, but bear in mind that code snippets in Arrow docs are processed by Arrow Playground (https://github.com/arrow-kt/arrow-playground/), so they might not be representative of what Ank generates (in terms of excessive new lines, that is).

Reply:
I checked both playground and non-playground snippets: https://next.arrow-kt.io/docs/arrow/core/listk/ Neither of them have a newline, however on a local run they had (the output in /docs/build/site that is).
I can take this out and open this in a new pr to get the other change in, because having multiline output rendered correcty is more important for me atm.

Reply:
Regarding https://next.arrow-kt.io/docs/quickstart/setup/, even if they are not executable, all Kotlin code (i.e. all code snippets marked as ```kotlin in the Markdown sources) in the Arrow docs is processed through Arrow Playground.

The only code snippets that Arrow Playground doesn't _stylize_ are other languages ones, e.g.:

https://next.arrow-kt.io/docs/quickstart/setup/

Reply:
While I can't check it locally, I think that what you said regarding the local output in `docs/build/site` having the newline at start falls in line with your fixes in the PR